/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2025. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package bus_service;

import "common.proto";

option go_package = "functionsystem/pkg/common/faas_common/grpc/pb/bus;bus";

// bus service provides APIs to runtime,
service BusService {
   // notify bus to connect driver
  rpc DiscoverDriver (DiscoverDriverRequest) returns (DiscoverDriverResponse) {}
}

message DiscoverDriverRequest {
  string     driverIP    = 1;
  string     driverPort  = 2;
  string     jobID       = 3;
  string     instanceID  = 4;
  string     functionName = 5;
}

message DiscoverDriverResponse {
  string     serverVersion = 1;
  string     posixPort     = 2;
  string     nodeId        = 3;
  string     hostIp        = 4;
}

message QueryInstanceRequest {
  string instanceID = 1;
}

message QueryInstanceResponse {
  common.ErrorCode code           = 1;
  string           message        = 2;
  string           status         = 3;
}
